src/track/beatmap.cpp

```diff
@@ -503,8 +502,14 @@ bool BeatMap::hasBeatInRange(audio::FramePos startPosition, audio::FramePos endP
             startPosition > endPosition) {
         return false;
     }
-    audio::FramePos beatPosition = findNextBeat(startPosition);
-    if (beatPosition <= endPosition) {
+    audio::FramePos beatPosition = findNextBeat(startPosition.toUpperFrameBoundary());
+
+    // FIXME: The following assertion should always hold true, but it doesn't,
+    // because the position matching in findNthBeat() is fuzzy. This should be
+    // resolved, and moved to the calling code, to only use fuzzy matches when
+    // actually desired.
+    // DEBUG_ASSERT(beatPosition >= startPosition.toUpperFrameBoundary());
```

Comment:
I guess the assertion in the comment should be`!beatPosition.isValid() || ...`?

Reply:
Yep, sure

Reply:
https://github.com/mixxxdj/mixxx/pull/4095/commits/3113dd962be4b2fc681ee253e4b9c65c7ebda308